codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -445,6 +434,21 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
         return new Result<String>(s, numberOfBytesConsumed);
     }
 
+    private static Result<byte[]> decodeByteArray(ByteBuf buffer) {
+        final Result<Integer> decodedSize = decodeMsbLsb(buffer);
+        int size = decodedSize.value;
+        int numberOfBytesConsumed = decodedSize.numberOfBytesConsumed;
+        if (size < 0) {
+            buffer.skipBytes(size);
+            numberOfBytesConsumed += size;
```

Comment:
This part of the code will be hit only in a hypothetical situation when the encoded bytebuf have some issues like length of a message field, say password is set to -1 etc.

However, if the assumption is, the encoded bytebuf is created out of the MqttEncoder, then this part of the code can be safely commented out.